### ISSUE TITLE
Problem with Synology and COPS

### ISSUE THREAD

**Issue by entuccio** [seblucas/cops#510]
Hello,
I've installed COPS inside my Synology Web server.
My Library is outside www default folder.

I set:

...
    $config['calibre_directory'] = '/volume1/Documenti/LibreriaPersonale/';
...

checkconfig.php said:

"Not all Calibre tables were found. Are you sure you're using the correct database."

I granted permission 777 to the library with ssh and "open_base" on PHP info is not set.

What can I do ?

Thanks

**Comment by SylvainDx** [comment]
On Synology you have to go in the php option and add the path of your library in the homedir field

Envoyé depuis Yahoo Mail pour Android 
 
  Le sam., janv. 8, 2022 à 12:53, ***@***.***> a écrit:   


Hello,
I've installed COPS inside my Synology Web server.
My Library is outside www default folder.

I set:

...
$config['calibre_directory'] = '/volume1/Documenti/LibreriaPersonale/';
...

checkconfig.php said:

"Not all Calibre tables were found. Are you sure you're using the correct database."

I granted permission 777 to the library with ssh and "open_base" on PHP info is not set.

What can I do ?

Thanks

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by SylvainDx** [comment]
Sorry I meant in webstation/php option / open basedir$

Envoyé depuis Yahoo Mail pour Android 
 
  Le sam., janv. 8, 2022 à 12:53, ***@***.***> a écrit:   


Hello,
I've installed COPS inside my Synology Web server.
My Library is outside www default folder.

I set:

...
$config['calibre_directory'] = '/volume1/Documenti/LibreriaPersonale/';
...

checkconfig.php said:

"Not all Calibre tables were found. Are you sure you're using the correct database."

I granted permission 777 to the library with ssh and "open_base" on PHP info is not set.

What can I do ?

Thanks

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by entuccio** [comment]
Works perfectly. Thank you :-)

**Comment by khiryos** [comment]
Hello,
I followed the whole installation process but still can't figure out why COPS does not work ("https://DDNS/cops" displayed nothing)

1) I have a web service portal running Apache 2.4 and PHP 7.6 (with all extensions checked)
2) My personal Calibre library is located at ":/volume1/books", and this path is written in the "config_local.php" file and in the PHP profile of Web Station
3) Read/Write permissions for http group are checked for the "books" folder
4) My firewall is disactivated (just for testing)

Do you have an idea of what I am doing wrong, please?

**Comment by marioscube** [comment]
Just for testing purposes:

Put your calibre library in a subdirectory of COPS (./COPS/books) and point to there in config_local.php.

Does this work?

**Comment by khiryos** [comment]
Hello Mario,

Thanks for helping. I tested what you suggest (and also switched to a full-stack IPv4 in order to use COPS on the internet, but it was another problem). Now I have a blank page with "No input file specified."

**Comment by marioscube** [comment]
So COPS works, Synology works, great!

Just put a password on COPS for external access.

I cannot test it, but COPS should work with IPv6.
Is there a reason why it should not?

**Comment by khiryos** [comment]
You are right, Mario. Now I have a user/password authentication but it is also enabled for local access (it's not a big deal for me). I don't know if COPS also works with IPv6.

**Comment by marioscube** [comment]
@khiryos 

1 - password
 Check out this info (haven't tried it myself yet but...) [https://stackoverflow.com/questions/4913000/use-htaccess-to-restrict-external-access-to-my-intranet](url)

2 - port 443
 Security through obscurity might just work for a home-server.
 I use a port number > 10.000 and forward this to 80 or 443 in my router. If I decide to change the port I only have to change the rules in my router and not all kinds of configuration files (I run multiple "servers" on several Pi's for testing-fun).

3 - IPv6
 Don't know. I have a tunnel IPv4 to IPv6 but have not tested much with it.

**Comment by khiryos** [comment]
> 

1 - Your link sends me to "No results matched your search."
2 - It's a good advice. I already did that for my default http/https ports (instead of 5000/5001 that Synology put) but I wonder how to change the port for COPS. It's not a straightforward way (click, enter, port, save). How to achieve the port change, please?
3 - Okay.

Thanks for your feedback.

**Comment by marioscube** [comment]
This should work (I hope):

[https://stackoverflow.com/questions/4913000/use-htaccess-to-restrict-external-access-to-my-intranet](https://stackoverflow.com/questions/4913000/use-htaccess-to-restrict-external-access-to-my-intranet)

**Comment by khiryos** [comment]
Hello Mario,
Yes, I know how to redirect ports in my router, but I wonder how to change the access port of the COPS application itself. When I enter https://DDNS:XXXX/cops, with XXXX redirected to 443 from my router settings, it does not work anymore.
Thanks for your link. I think the answer is here: https://tomschlick.com/conditional-htpasswd-multi-environments/ but I admit that don't have the strength to dive into lines of code.

**Comment by marioscube** [comment]
COPS has no settings for setting an access port.

If https://DDNS:443/cops worked before (https:///DDNS/cops) then https://DDNS:12345/cops should work if you forwarded port 12345 to 443 for the internal IP if COPS in your router.

**Comment by khiryos** [comment]
Now it works. You know why? Because I just had to put an "/" at the of the URL ^^

**Comment by marioscube** [comment]
Sometimes an “/“ is all it needs.........

It seems to me its browser dependent. Some browser add an “/“ automatically and some don’t. I sometimes have “//“ at the end of the url, and then it sometimes works, and sometimes not. Pretty frustrating. ;-)

**Comment by khiryos** [comment]
Aha you're right! Thanks again for your help ;) Have a nice weekend.

**Comment by Tos26** [comment]
Hello,

I use COPs for several years. I'm worried about the date of last commit.
Is the project still maintained or the developpers are gone ?
